azure/azure-dls-gen2/src/test/java/org/talend/components/adlsgen2/output/AdlsGen2OutputTest.java

```diff
@@ -0,0 +1,133 @@
+import org.talend.components.adlsgen2.runtime.output.BlobWriterFactory;
+import org.talend.components.adlsgen2.service.AdlsGen2Service;
+import org.talend.sdk.component.api.record.Record;
+import org.talend.sdk.component.api.service.record.RecordBuilderFactory;
+import org.talend.sdk.component.runtime.record.RecordImpl;
+
+class AdlsGen2OutputTest {
+
+    private AdlsGen2Output sut;
+
+    private OutputConfiguration outputConfiguration;
+
+    private MockedStatic<BlobWriterFactory> mockedStatic;
+
+    private RecordBuilderFactory recordBuilderFactoryMock;
+
+    @BeforeEach
+    void setUp() {
+        mockedStatic = Mockito.mockStatic(BlobWriterFactory.class);
+        outputConfiguration = new OutputConfiguration();
+        outputConfiguration.setDataSet(new AdlsGen2DataSet());
+        AdlsGen2Service serviceMock = Mockito.mock();
+        recordBuilderFactoryMock = Mockito.mock();
+        Mockito.when(recordBuilderFactoryMock.newRecordBuilder()).thenReturn(new RecordImpl.BuilderImpl());
+        JsonBuilderFactory jsonBuilderFactoryMock = Mockito.mock();
+        sut = new AdlsGen2Output(outputConfiguration, serviceMock, recordBuilderFactoryMock, jsonBuilderFactoryMock);
+    }
+
+    @Test
+    void testNormalFlow() throws Exception {
+        Record testRecord = recordBuilderFactoryMock.newRecordBuilder().withString("key", "value").build();
+        BlobWriter writerMock = Mockito.mock();
+        mockedStatic.when(() -> BlobWriterFactory.getWriter(Mockito.any(),
+                Mockito.any(), Mockito.any(), Mockito.any())).thenReturn(writerMock);
+
+        sut.init();
+        sut.beforeGroup();
+        sut.onNext(testRecord);
+        sut.afterGroup();
+        sut.release();
+
+        Mockito.verify(writerMock).writeRecord(testRecord);
+        Mockito.verify(writerMock).complete();
+    }
+
+    @Test
+    void testOnNextSkipEmptyRecords() {
+        Record emptyRecord = recordBuilderFactoryMock.newRecordBuilder().build();
+        Record testRecord = recordBuilderFactoryMock.newRecordBuilder().withString("key", "value2").build();
+        BlobWriter writerMock = Mockito.mock();
+        mockedStatic.when(() -> BlobWriterFactory.getWriter(Mockito.any(),
+                Mockito.any(), Mockito.any(), Mockito.any())).thenReturn(writerMock);
+
+        sut.init();
+        sut.beforeGroup();
+        sut.onNext(testRecord);
+        sut.onNext(emptyRecord);
+        sut.afterGroup();
+        sut.release();
+
+        Mockito.verify(writerMock, Mockito.times(1)).writeRecord(Mockito.any());
+    }
+
+    @Test
+    void testGetReaderThrowsExceptionOnInit() {
+        mockedStatic.when(() -> BlobWriterFactory.getWriter(Mockito.any(),
+                Mockito.any(), Mockito.any(), Mockito.any())).thenThrow(RuntimeException.class);
+
+        Assertions.assertThrows(AdlsGen2RuntimeException.class, () -> sut.init());
+    }
+
+    @Test
+    void testGetReaderThrowsExceptionOnFlush() throws Exception {
```

Comment:
testGetReaderThrowsExceptionOnFlush
```suggestion
    void testGetWriterThrowsExceptionOnFlush() throws Exception {
```